File: app/src/main/java/com/example/kursovayatesty/ShowQrActivity.java
package com.example.kursovayatesty;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.ImageButton;

import androidx.appcompat.app.AppCompatActivity;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.WriterException;
import com.google.zxing.qrcode.QRCodeWriter;

public class ShowQrActivity extends AppCompatActivity {

    /**
     * Метод жизненного цикла активности.
     * Получает строку с содержимым для QR-кода из Intent.
     * Создаёт QR-код и отображает его в ImageView.
     * Также настраивает кнопку закрытия.
     *
     * @param savedInstanceState - сохранённое состояние активности (не используется)
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_show_qr); // Устанавливаем разметку

        ImageView qrImage = findViewById(R.id.qrImageView);
        ImageButton closeButton = findViewById(R.id.closeButton);

        // Получаем текст для генерации QR-кода из переданного Intent
        String content = getIntent().getStringExtra("qr_content");
        if (content != null) {
            QRCodeWriter writer = new QRCodeWriter();
            try {
                // Генерируем битовую матрицу QR-кода, преобразуем в Bitmap и показываем в ImageView
                Bitmap bitmap = toBitmap(writer.encode(content, BarcodeFormat.QR_CODE, 800, 800));
                qrImage.setImageBitmap(bitmap);
            } catch (WriterException e) {
                e.printStackTrace(); // Ошибка генерации QR-кода
            }
        }

        // Обработчик кнопки закрытия: завершает активность
        closeButton.setOnClickListener(v -> finish());
    }

    /**
     * Преобразует BitMatrix (матрицу черно-белых пикселей) в объект Bitmap.
     *
     * @param matrix - битовая матрица QR-кода, где true - черный пиксель, false - белый.
     * @return объект Bitmap с изображением QR-кода.
     */
    private Bitmap toBitmap(com.google.zxing.common.BitMatrix matrix) {
        int width = matrix.getWidth();
        int height = matrix.getHeight();
        Bitmap bmp = Bitmap.createBitmap(width, height, Bitmap.Config.RGB_565);

        // Проходим по всем пикселям матрицы и устанавливаем цвет пикселя в Bitmap
        for (int x = 0; x < width; x++) {
            for (int y = 0; y < height; y++) {
                bmp.setPixel(x, y, matrix.get(x, y) ? 0xFF000000 : 0xFFFFFFFF); // Черный или белый
            }
        }
        return bmp;
    }
}

File: app/src/main/java/com/example/kursovayatesty/ShowQrActivity.kt
package com.example.kursovayatesty

import android.graphics.Bitmap
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.google.zxing.BarcodeFormat
import com.google.zxing.WriterException
import com.google.zxing.common.BitMatrix
import com.google.zxing.qrcode.QRCodeWriter

class ShowQrActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_show_qr)

        val qrImage = findViewById<ImageView>(R.id.qrImageView)
        val closeButton = findViewById<ImageButton>(R.id.closeButton)

        val content = intent.getStringExtra("qr_content")
        if (content != null) {
            val writer = QRCodeWriter()
            try {
                val bitmap = toBitmap(writer.encode(content, BarcodeFormat.QR_CODE, 800, 800))
                qrImage.setImageBitmap(bitmap)
            } catch (e: WriterException) {
                e.printStackTrace()
            }
        }

        closeButton.setOnClickListener { finish() }
    }

    private fun toBitmap(matrix: BitMatrix): Bitmap {
        val width = matrix.width
        val height = matrix.height
        val bmp = Bitmap.createBitmap(width, height, Bitmap.Config.RGB_565)

        for (x in 0..<width) {
            for (y in 0..<height) {
                bmp.setPixel(x, y, if (matrix[x, y]) -0x1000000 else -0x1)
            }
        }
        return bmp
    }
}
